clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10852,6 +10852,8 @@ void SYCLPostLink::ConstructJob(Compilation &C, const JobAction &JA,
   std::string OutputArg = Output.getFilename();
   if (T.getSubArch() == llvm::Triple::SPIRSubArch_gen && Device.data())
     OutputArg = ("intel_gpu_" + Device + "," + OutputArg).str();
+  else if (T.getSubArch() == llvm::Triple::SPIRSubArch_x86_64)
+    OutputArg = "spir64_x86_64," + OutputArg;
```

Comment:
Can you add a driver specific test that verifies the sycl-post-link arguments here?

Reply:
[48b7c7b](https://github.com/intel/llvm/pull/14590/commits/48b7c7ba4bc113ba1ad4b000ae6de1000c9b3034)